core/src/main/java/org/bitcoinj/wallet/DeterministicKeyChain.java

```diff
@@ -106,20 +106,33 @@
     @Nullable private DeterministicKey rootKey;
     @Nullable private DeterministicSeed seed;
     @Nullable private final ImmutableList<ChildNumber> accountPath;
+    private final Script.ScriptType outputScriptType;
 
     // Paths through the key tree. External keys are ones that are communicated to other parties. Internal keys are
     // keys created for change addresses, coinbases, mixing, etc - anything that isn't communicated. The distinction
     // is somewhat arbitrary but can be useful for audits. The first number is the "account number" but we don't use
     // that feature yet. In future we might hand out different accounts for cases where we wish to hand payers
     // a payment request that can generate lots of addresses independently.
     // The account path may be overridden by subclasses.
+    // m / 0'
     public static final ImmutableList<ChildNumber> ACCOUNT_ZERO_PATH = ImmutableList.of(ChildNumber.ZERO_HARDENED);
+    // m / 1'
+    public static final ImmutableList<ChildNumber> ACCOUNT_ONE_PATH = ImmutableList.of(ChildNumber.ONE_HARDENED);
     // m / 44' / 0' / 0'
     public static final ImmutableList<ChildNumber> BIP44_ACCOUNT_ZERO_PATH = ImmutableList.of(new ChildNumber(44, true),
             ChildNumber.ZERO_HARDENED, ChildNumber.ZERO_HARDENED);
     public static final ImmutableList<ChildNumber> EXTERNAL_SUBPATH = ImmutableList.of(ChildNumber.ZERO);
     public static final ImmutableList<ChildNumber> INTERNAL_SUBPATH = ImmutableList.of(ChildNumber.ONE);
 
+    private static ImmutableList<ChildNumber> defaultAccountPathFor(Script.ScriptType outputScriptType) {
```

Comment:
Why is this? Is it for bip49?

Reply:
No. BIP49 is out of scope, as it deals with P2SH-P2WPKH only.

Reply:
If you look at the implementation of the method, there is one path for P2PKH (the default old one) and another for the new P2WPKH output script type.

Reply:
@lontivero Also see https://github.com/bitcoinj/bitcoinj/pull/1563#issuecomment-462271008; I moved the method into a pluggable interface.